onchain/src/art_peace.cairo

```diff
@@ -990,66 +990,53 @@ pub mod ArtPeace {
-
+        let daily_new_colors_count = self.daily_new_colors_count.read();
         let day = self.day_index.read();
-        let next_day = day + 1;
         let votable_colors_count = self.votable_colors_count.read(day);
 
         let mut max_scores: Felt252Dict<u32> = Default::default();
-        let mut max_index: u8 = 0;
-
         let mut votable_index: u8 = 1; // 0 means no vote
-
         while votable_index <= votable_colors_count {
-            max_index = 0;
-            let mut index = votable_index;
-            let mut vote = self.color_votes.read((index, day));
+            let vote = self.color_votes.read((votable_index, day));
+            if vote <= max_scores.get(daily_new_colors_count.into() - 1) {
+                votable_index += 1;
+                continue;
+            }
             // update max scores if needed
-            while max_index < nb_max {
-                let mut m: felt252 = max_index.into();
-                if (max_scores.get(m) < vote) {
-                    let mut old_vote = max_scores.get(m);
-                    max_scores.insert(m, vote);
-                    vote = old_vote;
-
-                    max_index += 1;
-                    // update max dict
-                    while max_index < nb_max {
-                        m = max_index.into();
-                        old_vote = max_scores.get(m);
-                        max_scores.insert(m, vote);
-                        vote = old_vote;
-                        max_index += 1;
+            let mut max_scores_index: u32 = 0;
+            while max_scores_index < daily_new_colors_count {
+                if max_scores.get(max_scores_index.into()) < vote {
+                    // shift scores
+                    let mut i: u32 = daily_new_colors_count - 1;
+                    while i > max_scores_index {
+                        max_scores.insert(i.into(), max_scores.get(i.into() - 1));
+                        i -= 1;
                     };
-                };
-                max_index += 1;
+                    max_scores.insert(max_scores_index.into(), vote);
+                    break;
+                }
+                max_scores_index += 1;
             };
             votable_index += 1;
         };
 
         // find threshold
         let mut threshold: u32 = 0;
-        max_index = nb_max;
-        while max_index > 0 {
-            threshold = max_scores.get(max_index.into());
-            if threshold > 0 {
-                break;
-            }
-            max_index -= 1;
-        };
-
-        if threshold == 0 {
-            threshold = max_scores.get(0);
-        }
+        let mut min_index = daily_new_colors_count;
+        while threshold == 0
+            && min_index > 0 {
+                min_index -= 1;
+                threshold = max_scores.get(min_index.into());
+            };
         if threshold == 0 {
-            // no vote
-            threshold = 1;
+            // No votes
+            return;
```

Comment:
`threshold = 1` was to ensure than `votable_colors` will be set for the next day.
With a `return` here, `votable colors` will not be set.

Here is a test case for checking *no vote* behaviour:
```cairo
#[test]
fn no_vote_test() {
    let art_peace_address = deploy_contract();
    let art_peace = IArtPeaceDispatcher { contract_address: art_peace_address };
    let previous_votable_colors = art_peace.get_votable_colors();
    let previous_palette = art_peace.get_colors();    
    snf::start_warp(CheatTarget::One(art_peace_address), DAY_IN_SECONDS);
    art_peace.increase_day_index();
    let votable_colors = art_peace.get_votable_colors();
    let palette = art_peace.get_colors();
    assert_eq!(votable_colors, previous_votable_colors, "Votable colors mismatch");
    assert_eq!(palette, previous_palette, "Palette colors mismatch");
    snf::stop_warp(CheatTarget::One(art_peace_address));
}
```